### ISSUE TITLE
patch should create files (and folders) if the don't already exist

### ISSUE THREAD

**Issue by GoogleCodeExporter** [techtonik/python-patch#3]
```
What steps will reproduce the problem?
1. attempt to run a patch where some of the hunks are non-existing files on the 
target

What is the expected output? What do you see instead?
patch should create these files - as does the linux patch utility, possibly 
creating directories as well

What version of the product are you using? On what operating system?


Please provide any additional information below.


```

Original issue reported on code.google.com by `dlr...@gmail.com` on 12 Aug 2009 at 4:43
* Blocking: #14
* Blocked on: #13

**Comment by techtonik** [comment]
Duplicate of #28 (which has patch attached).
